### PR TITLE
refactor: match upstream print preview handling

### PR DESCRIPTION
**Pull request by codebytere**
#### Description of Change

Printing refactor to more closely match upstream upstream. Once this is merged to the master roll we can cherry-pick it to 10-x-y

cc @deepak1556 

Notes: none

### CODE REVIEW

File: shell/browser/printing/print_preview_message_handler.cc [hunk 1]
```diff
@@ -107,14 +112,45 @@ void PrintPreviewMessageHandler::OnMetafileReadyForPrinting(
   }
 }
 
+void PrintPreviewMessageHandler::OnPrepareForDocumentToPdfDone(
+    const PrintHostMsg_PreviewIds& ids,
+    printing::mojom::PrintCompositor::Status status) {
+  DCHECK_CURRENTLY_ON(BrowserThread::UI);
+  if (status != printing::mojom::PrintCompositor::Status::kSuccess) {
+    LOG(ERROR) << "Preparing document for pdf failed with error " << status;
```
Comment (deepak1556) [inline shell/browser/printing/print_preview_message_handler.cc:120]: Will `OnMetafileReadyForPrinting` be called even when this fails ? want to make sure we don't have an unresolved promise.

Comment (codebytere) [reply shell/browser/printing/print_preview_message_handler.cc:120]: yes - this is always triggered from local testing. if we try and reject before this, in page composition failure, it results in duplicate rejections

**Comment by deepak1556** [comment]
Good to see a patch go away.

**Comment by codebytere** [comment]
merging as the failure:
> AssertionError: is not alwaysOnTop: expected false to be true

is unrelated.

**Pull request merged**